### ISSUE TITLE
Nested try/catch statements trigger the no-unreachable rule incorrectly

### ISSUE THREAD

**Issue by antca** [denoland/deno_lint#582]
The following code produces the issue:

```js
try {
  try {
    throw new Error();
  } catch {
    throw new Error();
  }
} catch {
  console.log("Statement reached!");
}
```

deno 1.6.3 (release, x86_64-unknown-linux-gnu)
v8 8.8.294
typescript 4.1.3

**Comment by magurotuna** [comment]
Thanks for the report, I'll look into it.

**Comment by magurotuna** [comment]
I've found where the cause is, but it will take some time to fix it. I hope I'll make it before the next release of Deno.

**Comment by antca** [comment]
Thanks for your work on this. It is not something blocking me in any way.
I would like to help but I don't know rust at all (yet).
